### DISCUSSION TITLE
SQ lock ?

### DISCUSSION THREAD

**Discussion by AaronJowitt** [PE5PVB/TEF6686_ESP32#437] [Ideas]
Not so much for me, but noticed a couple of discord user comments about often accidentally knocking the SQ knob given its close position between rotary & vol.

What about a 'SQ lock' feature?  

When enabling - it stores the current SQ knob value to eeprom, and the radio permanently uses that stored value as a sq override, ignoring all subsequent turns of the SQ knob.

When disabling - the override value is deleted and the SQ knob operates as normal.

**Comment by ohmytime** [comment]
I think it can be done, waiting for others to discuss
In addition, is it necessary to  prompting the status of SQ, frozen or non-frozen on the main screen ?

**Reply by AaronJowitt** [reply]
When frozen/locked; 

the stored SQ value should be on screen, as normal.  Perhaps permanently greyed out? 
Or alternatively, whenever the user is rotating the SQ knob, temporarily grey out the SQ or temporarily show a lock symbol in its place.

**Comment by Yurij20** [comment]
Yes, it’s better to do it, sometimes I accidentally touch the middle encoder with my hand.  I don't need this function itself.

**Comment by Overland-DX** [comment]
I physically removed the squelch knob because of that, so I think it would be a useful feature.

**Comment by PE5PVB** [comment]
I changed "Show squelch" into "Use squelch". When disabled, squelch can't be set. You can find the option in Display Settings.

**Reply by AaronJowitt** [reply]
That's an improvement. You are disabling the SQ knob and setting SQ to OFF. 

What would be better & more useful, is if you only disable the SQ knob. So the SQ remains active with the value (in eeprom) that was used at the time the SQ was 'locked'.

For example, assume I currently have the SQ set to 5 dBuV (or OFF, or ST or 20 dBuV or whatever) and that's the SQ value i like to use and lock at.  I enable the lock & the radio will then persistently use (and display) SQ 5, and disable the SQ knob.

**Reply by PE5PVB** [reply]
Yes, that part will be moved to future development. My idea is to add a extra menu item where you can set a fixed squelch level or use potmeter.

**Reply by AaronJowitt** [reply]
Sure, that would work but make the "Use squelch" option largely redundant.  

i.e. if you later add a "Fixed squelch" option that disabled the SQ knob and can be set to any valid SQ value, you replicate the "Use squelch" function if the SQ is set/fixed to a value of OFF.

I wonder if you instead had one simple SQ function "Lock squelch" that only disabled the SQ knob (and kept SQ active/fixed at its last used value), you would cover all the different use cases in the one function.

**Reply by PE5PVB** [reply]
We will see, but not for now.

**Reply by AaronJowitt** [reply]
ok, sounds good, thank you Sjef.  When you look at SQ again later, suggest SQ config may be better placed in Audio settings than Display settings.
